contracts/treasury/contracts/src/ZrxTreasury.sol

```diff
@@ -0,0 +1,381 @@
+    /// @param proposalId The ID of the proposal to vote on.
+    /// @param support Whether to support the proposal or not.
+    /// @param operatedPoolIds The pools operated by `account`. The
+    ///        ZRX currently delegated to those pools will be accounted
+    ///        for in the voting power.
+    function castVote(
+        uint256 proposalId,
+        bool support,
+        bytes32[] memory operatedPoolIds
+    )
+        public
+        override
+    {
+        if (proposalId >= proposalCount()) {
+            revert("castVote/INVALID_PROPOSAL_ID");
+        }
+        if (hasVoted[proposalId][msg.sender]) {
+            revert("castVote/ALREADY_VOTED");
+        }
+
+        Proposal memory proposal = proposals[proposalId];
+        if (
+            proposal.voteEpoch != stakingProxy.currentEpoch() ||
+            _hasVoteEnded(proposal.voteEpoch)
+        ) {
+            revert("castVote/VOTING_IS_CLOSED");
+        }
+
+        uint256 votingPower = getVotingPower(msg.sender, operatedPoolIds);
+        if (votingPower == 0) {
+            revert("castVote/NO_VOTING_POWER");
+        }
+
+        if (support) {
+            proposals[proposalId].votesFor = proposals[proposalId].votesFor
+                .safeAdd(votingPower);
+            hasVoted[proposalId][msg.sender] = true;
+        } else {
+            proposals[proposalId].votesAgainst = proposals[proposalId].votesAgainst
+                .safeAdd(votingPower);
+            hasVoted[proposalId][msg.sender] = true;
+        }
+
+        emit VoteCast(
+            msg.sender,
+            operatedPoolIds,
+            proposalId,
+            support,
+            votingPower
+        );
+    }
+
+    /// @dev Executes a proposal that has passed and is
+    ///      currently executable.
+    /// @param proposalId The ID of the proposal to execute.
+    /// @param actions Actions associated with the proposal to execute.
+    function execute(uint256 proposalId, ProposedAction[] memory actions)
+        public
+        payable
+        override
+    {
+        if (proposalId >= proposalCount()) {
+            revert("execute/INVALID_PROPOSAL_ID");
+        }
+        Proposal memory proposal = proposals[proposalId];
+        _assertProposalExecutable(proposal, actions);
+
+        proposals[proposalId].executed = true;
+
+        for (uint256 i = 0; i != actions.length; i++) {
+            ProposedAction memory action = actions[i];
+            (bool didSucceed, ) = action.target.call{value: action.value}(action.data);
+            require(
+                didSucceed,
+                "execute/ACTION_EXECUTION_FAILED"
+            );
+        }
+
+        emit ProposalExecuted(proposalId);
+    }
+
+    function castVoteBySig(
+        uint256 proposalId,
+        bool support,
+        bytes32[] memory operatedPoolIds,
+        LibSignature.Signature memory signature
+    )
+        public
+    {
+        revert("NOT_IMPLEMENTED");
```

Comment:
Why do we have these if they're not going to be implemented?

Reply:
was aspirational